packages/client/src/generation/utils.ts

```diff
@@ -138,6 +140,12 @@ export function getModelArgName(modelName: string, action?: DMMF.ModelAction): s
       return `${modelName}FindRawArgs`
     case DMMF.ModelAction.aggregateRaw:
       return `${modelName}AggregateRawArgs`
+    case 'findFirstOrThrow':
+      return `${modelName}FindFirstOrThrowArgs`
```

Comment:
Can we reuse the args from the real findFirst here instead of generating the same types again?

Reply:
The problem here is that `rejectOnNotFound` right now is included into `FindFirstArgs`. I agree that generating huge type just because of one field is not very optimal. The cleanest way to fix it, in my opinion, would be:

- Define base args type, say, `FindFirstBaseArgs` which will include all arguments minus `rejectOnNotFound`
- Define `FindFirstArgs` as `FindFirstBaseArgs & { rejectOnNotFound }`
- Define `FindFirstArgs` as a type alias for `FindFirstBaseArgs`.

Last option will leave us some wiggle room in case in future we'd want to have some arguments specific for `OrThrow` variants.  If you think that's premature, we can ditch base type altogether and simple define `FindFirstArgs` as `FindFirstOrThrowArgs & { rejectOnNotFound }`. WDYT?

Reply:
I'm happy with both, though the first proposal feels the cleanest.

Reply:
One thing to consider is in the future we'll be dropping `rejectOnNotFound` completely, which will make both types identical.

It might be important to keep this in mind when choosing between the approaches.

Reply:
@aqrln @millsp Updated, implemented first approach. When we will be removing `rejectOnNotFound`, either both args types could be an alias for the same type (in case we want distinct names for consitency) or we'll skip `OrThrow` generation completely and just re-use them args generated by base variant.